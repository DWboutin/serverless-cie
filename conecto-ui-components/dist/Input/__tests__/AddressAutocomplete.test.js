"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _Input = require("../../Input");

global.Element = {};
global.console = {
  error: jest.fn()
};
const onPressEnter = jest.fn();
const onKeyDown = jest.fn();
const onChange = jest.fn();
const onBlur = jest.fn();

const setupGoogleMock = () => {
  /*** Mock Google Maps JavaScript API ***/
  const google = {
    maps: {
      places: {
        AutocompleteService: function () {},
        PlacesServiceStatus: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS'
        }
      },
      Geocoder: () => {},
      GeocoderStatus: {
        ERROR: 'ERROR',
        INVALID_REQUEST: 'INVALID_REQUEST',
        OK: 'OK',
        OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
        REQUEST_DENIED: 'REQUEST_DENIED',
        UNKNOWN_ERROR: 'UNKNOWN_ERROR',
        ZERO_RESULTS: 'ZERO_RESULTS'
      }
    }
  };
  global.window.google = google;
};

describe('AddressAutocomplete', () => {
  beforeEach(() => {
    setupGoogleMock();
    global.console.error.mockClear();
    onPressEnter.mockClear();
    onKeyDown.mockClear();
    onChange.mockClear();
    onBlur.mockClear();
  });
  it('should match snapshot', () => {
    const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Input.AddressAutocomplete, {
      id: "testField",
      name: "testField",
      type: "text",
      label: "Address"
    }));
    expect(wrapper).toMatchSnapshot();
  });
  it('should match mounted snapshot', () => {
    const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Input.AddressAutocomplete, {
      id: "testField",
      name: "testField",
      type: "text",
      label: "Address"
    }));
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('should have the "focused" class when focus on input', () => {
    const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Input.AddressAutocomplete, {
      id: "testField",
      name: "testField",
      type: "text",
      label: "Address"
    }));
    wrapper.find('input').last().simulate('focus');
    expect(wrapper.render().hasClass('focused')).toEqual(true);
    wrapper.unmount();
  });
  it('should handle Enter key press correctly', () => {
    const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Input.AddressAutocomplete, {
      id: "testField",
      name: "testField",
      type: "text",
      label: "Address",
      onPressEnter: onPressEnter
    }));
    wrapper.find('input').last().simulate('keyDown', {
      keyCode: 13
    });
    expect(onPressEnter).toBeCalled();
    wrapper.unmount();
  });
  it('should handle Key Down correctly', () => {
    const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Input.AddressAutocomplete, {
      id: "testField",
      name: "testField",
      type: "text",
      label: "Address",
      onKeyDown: onKeyDown
    }));
    wrapper.find('input').last().simulate('keyDown', {
      keyCode: 1
    });
    expect(onKeyDown).toBeCalled();
    wrapper.unmount();
  });
});