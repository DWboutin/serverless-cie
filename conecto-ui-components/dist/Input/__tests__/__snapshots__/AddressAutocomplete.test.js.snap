// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressAutocomplete should match mounted snapshot 1`] = `
<AddressAutocomplete
  debounce={450}
  id="testField"
  label="Address"
  name="testField"
  type="text"
>
  <div
    className="to-address-autocomplete"
    id="to-address-autocomplete-testField"
  >
    <PlacesAutocomplete
      debounce={450}
      highlightFirstSuggestion={false}
      onBlur={[Function]}
      onChange={[Function]}
      onError={[Function]}
      onSelect={[Function]}
      searchOptions={
        Object {
          "componentRestrictions": Object {
            "country": "ca",
          },
        }
      }
      shouldFetchSuggestions={false}
      value=""
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        aria-label="Address"
        aria-labelledby="to-address-autocomplete-testField-label"
        autoComplete="autocomplete_off_1234567890"
        className="to-input-field"
        debounce={450}
        disabled={false}
        id="testField"
        name="testField"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="combobox"
        type="text"
        value=""
      />
      <Label
        className="to-input-label"
        htmlFor="testField"
        id="to-address-autocomplete-testField-label"
      >
        <label
          className="to-input-label"
          htmlFor="testField"
          id="to-address-autocomplete-testField-label"
        >
          Address
        </label>
      </Label>
      <div
        className="to-address-autocomplete-choices"
      >
        <ul />
      </div>
    </PlacesAutocomplete>
  </div>
</AddressAutocomplete>
`;

exports[`AddressAutocomplete should match snapshot 1`] = `
<div
  className="to-address-autocomplete"
  id="to-address-autocomplete-testField"
>
  <PlacesAutocomplete
    debounce={450}
    highlightFirstSuggestion={false}
    onBlur={[Function]}
    onChange={[Function]}
    onError={[Function]}
    onSelect={[Function]}
    searchOptions={
      Object {
        "componentRestrictions": Object {
          "country": "ca",
        },
      }
    }
    shouldFetchSuggestions={false}
    value=""
  >
    <Component />
  </PlacesAutocomplete>
</div>
`;
